### ISSUE TITLE
[ML] Error - Random Variable parameter was -nan, but must be > 0

### ISSUE THREAD

**Issue by dolaru** [elastic/ml-cpp#271]
Spotted in 6.5.0

The following error messages were found in ES logs while running the `gallery1` QA dataset:
```
[907][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 335, d2 = 333, x = -nan
[910][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 280, d2 = 278, x = -nan
[911][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 271, d2 = 269, x = -nan
[913][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 42, d2 = 40, x = -nan | repeated [4]
[914][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 21, d2 = 19, x = -nan
[915][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 42, d2 = 40, x = -nan
[916][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 25, d2 = 23, x = -nan
[918][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 21, d2 = 19, x = -nan | repeated [4]
[919][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 30, d2 = 28, x = -nan
[920][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 335, d2 = 333, x = -nan
[921][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 21, d2 = 19, x = -nan
[922][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 52, d2 = 50, x = -nan
[923][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 335, d2 = 333, x = -nan
[925][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 126, d2 = 124, x = -nan | repeated [8]
[926][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 41, d2 = 39, x = -nan
[927][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 23, d2 = 21, x = -nan
[929][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 335, d2 = 333, x = -nan
[930][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 41, d2 = 39, x = -nan
[937][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [7DdayTu] [gallery1_20181023-1031_650_2275] [autodetect/30069] [CStatisticalTests.cc@99] Failed to compute significance Error in function boost::math::cdf(fisher_f_distribution const&, double): Random Variable parameter was -nan, but must be > 0 ! d1 = 47, d2 = 45, x = -nan
```

Analysis config:
```json
{
  "bucket_span": "1h",
  "influencers": [
    "clientip", "file"
  ],
  "detectors": [
    {
      "over_field_name": "clientip",
      "by_field_name": "status",
      "function": "rare"
    },
    {
      "over_field_name": "clientip",
      "by_field_name": "uri",
      "function": "freq_rare"
    },
    {
      "over_field_name": "clientip",
      "by_field_name": "status",
      "function": "high_count"
    },
    {
      "over_field_name": "clientip",
      "by_field_name": "uri",
      "function": "high_count"
    },
    {
      "over_field_name": "clientip",
      "by_field_name": "method",
      "function": "sum",
      "field_name": "bytes"
    }
  ]
}
```

**Comment by edsavage** [comment]
These errors have been confirmed to have started occurring as a consequence of changes in #198. A quick-fix has been provided in #272 but a deeper analysis of the cause is required

**Comment by tveasey** [comment]
The root cause for this was fixed in #280.
